### PR TITLE
Added Type definitions for WordPress admin window global

### PR DESCRIPTION
**Pull request by seebeen**
- [* ] Use a meaningful title for the pull request. Include the name of the package modified.
- [* ] Test the change in your own code. (Compile and run.)
- [* ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [*] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [*] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [*] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [*] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [*] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [*] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [*] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [*] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

I'd like to point out that writing these definitions is insanely hard, `window.wp` object is split over multiple files, and all object properties are build via `Backbone.extend`, so I needed to use any in a lot of places in order to make it work.
This is a work in progress, but I consider it a solid base which enables people to work with `window.wp` global in TypeScript

### CODE REVIEW

File: types/wordpress__admin/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Type definitions for @wordpress/admin 5.8
```
Comment (andrewbranch) [inline types/wordpress__admin/index.d.ts:1]: Does this package exist? Where can I find the source code?

Comment (seebeen) [reply types/wordpress__admin/index.d.ts:1]: It doesn't. It's not an npm package, it's the wordpress-core itself which uses the defined objects.

Comment (andrewbranch) [reply types/wordpress__admin/index.d.ts:1]: See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a585d5fd2c4aaffae5c8e74e0763c7406689cdf1/types/activex-access/index.d.ts#L1 for an example of how to write this header, and remove the tslint npm-naming exception. (Ideally, remove all the tslint exceptions. Given that this is a new package, there should be no reason to start out with a bunch of rules disabled.)

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/wordpress__admin/index.d.ts (hunk 1 above)
Comment (seebeen) [reply types/wordpress__admin/index.d.ts]: Implemented changes per request!

**Review by andrewbranch — APPROVED** [review]

**Comment by seebeen** [comment]
Ready to merge

**Pull request merged**